src/QsCompiler/DocumentationGenerator/DocgenStep.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.Quantum.QsCompiler;
+using Microsoft.Quantum.QsCompiler.Experimental;
+using Microsoft.Quantum.QsCompiler.SyntaxTree;
+
+namespace Microsoft.Quantum.Documentation
+{
+    public class DocumentationGenerationStep : IRewriteStep
```

Comment:
Could you please add a minimal doc comment? And since none of the other steps are ending in *Step, it would be make sense to just call it DocumentationGeneration here, too.